### DISCUSSION TITLE
Can someone help me with this code. I'm trying to loop through line items. New to php

### DISCUSSION THREAD

**Discussion by Budrose** [XeroAPI/xero-php-oauth2#338] [General]
$guid = $result->getInvoices()[0]->getInvoiceID();
	echo $guid;
	
	$invoice = $accountingApi->getInvoice($xeroTenantId, $guid);
	echo $invoice;
	foreach($invoice->LineItems as $LineItem){
	   echo $code = $LineItem->ItemCode;
	   /* do something with the $code */
	}

But getting this error message

Notice: Undefined property: XeroAPI\XeroPHP\Models\Accounting\Invoices::$LineItems in C:\xampp\htdocs\xero-AlpineFreshOauth2\testUpdate.php on line 77

Warning: Invalid argument supplied for foreach() in C:\xampp\htdocs\xero-AlpineFreshOauth2\testUpdate.php on line 77

**Comment by calcinai** [comment]
Hi Brenda, I'd recommend checking that LineItems is set before trying to iterate it.  

P.S. I believe you ran into a similar issue back in 2015 when using xero-php, so check out that email thread where I provided some examples.

**Comment by Budrose** [comment]
Ah. Thanks Michael. I not programming much in PHP. Can you recommend a good
online course to do to become more proficient?


Brenda Walker
IT Consultant
***@***.***
021 910 975
